controller.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"flag"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/zalando-incubator/kube-ingress-aws-controller/aws"
+	"strconv"
+)
+
+var (
+	autoScalingGroupName string
+	securityGroupName    string
+	apiServerBaseURL     string
+	pollingInterval      uint
+)
+
+func waitForTerminationSignals(signals ...os.Signal) chan os.Signal {
+	c := make(chan os.Signal, 1)
+	for _, s := range signals {
+		signal.Notify(c, s)
+	}
+	return c
+}
+
+func loadEnviroment() error {
+	flag.Usage = usage
+	flag.StringVar(&autoScalingGroupName, "auto-scaling-group-name", "", "manually sets the auto scaling group name. "+
+		"if empty will try to resolve that using ec2 metadata")
+	flag.StringVar(&securityGroupName, "security-group-name", "", "sets the security group for the managed load balancers. "+
+		"if not specified the ASG tag `KubernestesCluster` is used to derive the SG as `<cluster>-worker-lb`")
+	flag.StringVar(&apiServerBaseURL, "api-server-base-url", "http://127.0.0.1:8001", "sets the kubernetes api server base url. "+
+		"if empty will try to use the common proxy url http://127.0.0.1:8001")
+	flag.UintVar(&pollingInterval, "polling-interval", 30, "sets the polling interval (in seconds) for ingress resources. "+
```

Comment:
Here you could also use https://golang.org/pkg/flag/#DurationVar then you can specify any interval, not only in seconds. e.g. `1m`.